google/resource_kms_crypto_key.go

```diff
@@ -111,6 +112,46 @@ func resourceKmsCryptoKeyCreate(d *schema.ResourceData, meta interface{}) error
 	return resourceKmsCryptoKeyRead(d, meta)
 }
 
+func resourceKmsCryptoKeyUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	cryptoKeyId, err := parseKmsCryptoKeyId(d.Id(), config)
+	if err != nil {
+		return err
+	}
+
+	key := cloudkms.CryptoKey{Purpose: "ENCRYPT_DECRYPT"}
+
+	if d.HasChange("rotation_period") && d.Get("rotation_period") != "" {
+		rotationPeriod := d.Get("rotation_period").(string)
+		nextRotation, err := kmsCryptoKeyNextRotation(time.Now(), rotationPeriod)
+
+		if err != nil {
+			return fmt.Errorf("Error setting CryptoKey rotation period: %s", err.Error())
+		}
+
+		key.NextRotationTime = nextRotation
+		key.RotationPeriod = rotationPeriod
+	}
+
+	cryptoKey, err := config.clientKms.Projects.Locations.KeyRings.CryptoKeys.Patch(
+		cryptoKeyId.cryptoKeyId(),
+		&key,
+	).
+		UpdateMask("rotation_period,next_rotation_time").
+		Do()
+
+	if err != nil {
+		return fmt.Errorf("Error updating CryptoKey: %s", err.Error())
+	}
+
+	log.Printf("[DEBUG] Updated CryptoKey %s", cryptoKey.Name)
+
+	d.SetId(cryptoKeyId.cryptoKeyId())
```

Comment:
I don't think this is entirely necessary (?), but I see this being done across various Terraform providers, so I suppose it can add some _guarantee_.